internal/daemon/controller/auth/auth.go

```diff
@@ -644,6 +647,9 @@ func (v verifier) performAuthCheck(ctx context.Context) (
 		aclResults.AuthenticationFinished = true
 		aclResults.Authorized = true
 		retErr = nil
+		grantTuples = []perms.GrantTuple{
+			{},
+		}
```

Comment:
Incredible, this must've been broken for almost a year (since we released pagination). I guess we never tried listing as the recovery user? Great spot!

Reply:
We test `ListSessions` with the recovery user but it worked due to `GrantHash` returning a slice of empty bytes